### ISSUE TITLE
Cannot authenticate via LDAP with :group set

### ISSUE THREAD

**Issue by OpenNebulaProject** [OpenNebula/one#1489]
---

Author Name: **Florent Peyraud** (Florent Peyraud)
Original Redmine Issue: 5559, https://dev.opennebula.org/issues/5559
Original Date: 2017-11-16

---

Hi
When trying to authenticate with LDAP using the :group parameter set to restrict the users allowed to log in, it fails with following logs

```
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 Command execution fail: /var/lib/one/remotes/auth/ldap/authenticate florent.peyraud - ****

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: Command execution fail: /var/lib/one/remotes/auth/ldap/authenticate florent.peyraud - ****
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 Trying server server 1

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: Trying server server 1
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 Exception raised authenticating to LDAP

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: Exception raised authenticating to LDAP
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 #<NoMethodError: undefined method `force_encoding' for nil:NilClass>

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: #<NoMethodError: undefined method `force_encoding' for nil:NilClass>
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 /usr/lib/one/ruby/opennebula/ldap_auth.rb:158:in `is_in_group?'

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: /usr/lib/one/ruby/opennebula/ldap_auth.rb:158:in `is_in_group?'
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 /var/lib/one/remotes/auth/ldap/authenticate:79:in `block in <main>'

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: /var/lib/one/remotes/auth/ldap/authenticate:79:in `block in <main>'
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 /var/lib/one/remotes/auth/ldap/authenticate:59:in `each'

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: /var/lib/one/remotes/auth/ldap/authenticate:59:in `each'
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 /var/lib/one/remotes/auth/ldap/authenticate:59:in `<main>'

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: /var/lib/one/remotes/auth/ldap/authenticate:59:in `<main>'
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 Could not authenticate user florent.peyraud

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: Could not authenticate user florent.peyraud
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: LOG I 8 ExitCode: 255

Thu Nov 16 17:34:00 2017 [Z0][AuM][I]: ExitCode: 255
Thu Nov 16 17:34:00 2017 [Z0][AuM][D]: Message received: AUTHENTICATE FAILURE 8 -

Thu Nov 16 17:34:00 2017 [Z0][AuM][E]: Auth Error: 
Thu Nov 16 17:34:00 2017 [Z0][ReM][D]: Req:9808 UID:-1 one.user.info invoked , -1
Thu Nov 16 17:34:00 2017 [Z0][ReM][E]: Req:9808 UID:- one.user.info result FAILURE [one.user.info] User couldn't be authenticated, aborting call.


```

**Comment by OpenNebulaProject** [comment]
---

Original Redmine Comment
Author Name: **Florent Peyraud** (Florent Peyraud)
Original Date: 2017-11-16T22:42:41Z

---

Fix proposed in pull request #580
CHeers

**Comment by vholer** [comment]
Proposed PR: https://github.com/OpenNebula/one/pull/1523
